inspirehep/modules/workflows/tasks/manual_merging.py

```
@@ -0,0 +1,159 @@
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Tasks related to manual merge generic record."""
+
+from __future__ import absolute_import, division, print_function
+
+import json
+
+from inspire_json_merger.inspire_json_merger import inspire_json_merge
+from inspire_dojson.utils import get_record_ref
+from inspirehep.modules.workflows.tasks.merging import _insert_wf_record_source
+from inspirehep.utils.record import get_source
+from inspirehep.utils.record_getter import get_db_record
+from inspirehep.modules.workflows.utils import with_debug_logging
+from invenio_db import db
+
+
+def _check_sources(wf_obj, head, update):
+    # Many records in Inspire don't have an acquisition_source.
+    # If the records to merge also don't, they are treated as arxiv records.
+    # We expect the `manual_merge` workflow to work on arxiv-publisher or
+    # publisher-arxiv records. In other cases, the arxive-arxiv strategy
+    # will be used, enforcing acquisition_source field to arxiv.
+    source_head = wf_obj.extra_data.get('head_source')
+    source_updated = wf_obj.extra_data.get('update_source')
+
+    if not head.get('acquisition_source'):
+        head['acquisition_source'] = {'source': source_head or 'arxiv'}
+
+    if not update.get('acquisition_source'):
+        update['acquisition_source'] = {'source': source_updated or 'arxiv'}
+
+
+def _get_head_and_update(obj):
+    """Retrieve from the database the records defined by their id in the
+    workflow object, in according to the pid_type value in `obj`.
+
+    Arguments:
+        obj(WorkflowObject): the current workflow object
+
+    Returns:
+        tuple(InspireRecord): the two records to merge
+
+    Raise:
+        RecordGetterError in case head or update does not exist
+    """
+    pid_type = obj.extra_data.get('pid_type')
+    recid_head = obj.extra_data.get('recid_head')
+    recid_update = obj.extra_data.get('recid_update')
```

if you expect the keys to be there, use `[]` and let it fail if they are not there.